File: Python/programmers/etc/level2/요격_시스템.py
def solution(targets):
    targets.sort(key=lambda t: t[1])
    answer = 0
    prev_e = 0
    for s, e in targets:
        if s >= prev_e:
            prev_e = e
            answer += 1
    return answer


targets = [[4,5],[4,8],[10,14],[11,13],[5,12],[3,7],[1,4]]
print(solution(targets)) # 3

File: Java/programmers/etc/level2/요격시스템.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 요격시스템 {
    public int solution(int[][] targets) {
        int answer = 0;
        Arrays.sort(targets, (o1, o2) -> {return o1[1]-o2[1];});
        int prev_e = 0;
        
        for (int i=0; i<targets.length; i++) {
            if (targets[i][0] >= prev_e) {
                prev_e = targets[i][1];
                answer++;
            }
        }
        
        return answer;
    }
}
